Orange/widgets/data/tests/test_owfeatureconstructor.py

```diff
@@ -71,6 +73,27 @@ def test_construct_variables_string(self):
                              str(data[i * 50, "iris"]) + "_name")
 
 
+GLOBAL_CONST = 2
+
+
+class PicklingTest(unittest.TestCase):
+    CLASS_CONST = 3
+
+    def test_lambdas_pickle(self):
+        NONLOCAL_CONST = 5
+
+        lambda_func = lambda x, LOCAL_CONST=7: \
+            x * LOCAL_CONST * NONLOCAL_CONST * self.CLASS_CONST * GLOBAL_CONST
+
+        def nested_func(x, LOCAL_CONST=7):
+            return x * LOCAL_CONST * NONLOCAL_CONST * self.CLASS_CONST * GLOBAL_CONST
+
+        self.assertEqual(lambda_func(11),
+                         pickle.loads(pickle.dumps(lambda_func))(11))
+        self.assertEqual(nested_func(11),
```

Comment:
Interestingly, it remembers the exact values of locals and nonlocals at pickling, so the computed result after unpickling is different. Changing the global after pickling errors with:
```
Traceback (most recent call last):
  File "/home/jk/PycharmProjects/orange3/Orange/widgets/data/tests/test_owfeatureconstructor.py", line 94, in test_lambdas_pickle
    pickled_lambda = pickle.dumps(lambda_func)
  File "/home/jk/Downloads/dill/dill/dill.py", line 256, in dumps
    dump(obj, file, protocol, byref, fmode, recurse)#, strictio)
  File "/home/jk/Downloads/dill/dill/dill.py", line 249, in dump
    pik.dump(obj)
  File "/usr/lib/python3.4/pickle.py", line 412, in dump
    self.save(obj)
  File "/usr/lib/python3.4/pickle.py", line 479, in save
    f(self, obj) # Call unbound method with explicit self
  File "/home/jk/Downloads/dill/dill/dill.py", line 1268, in save_function
    globs = globalvars(obj, recurse=True, builtin=True)
  File "/home/jk/Downloads/dill/dill/detect.py", line 200, in globalvars
    _vars = globalvars(obj.cell_contents, recurse, builtin) or {}
ValueError: Cell is empty
```
It works if a global is defined `global`, but this statement is impossible in lambdas. The difference `global` statement makes is in `LOAD_DEREF` vs `LOAD_GLOBAL` opcodes. I don't know if it is or will become an issue in a practical case.